From b7c9f8d5326ad4722ef74ef4477bc9014a278e6f Mon Sep 17 00:00:00 2001
From: Lucas de Ataides <lucas.deataidesbarreto@windriver.com>
Date: Mon, 4 Mar 2024 12:09:40 -0300
Subject: [PATCH] Add missing initial delay in readiness probes

After applying STX-Openstack, some API pods are failing to pass the
readiness probe. This change adds a initialDelaySeconds value to the
deployment files of the services that are having this warning.

The liveness probe for these deployments already have a
initialDelaySeconds of 30 seconds, adding this value to the readiness
probe supresses this issue.

Signed-off-by: Lucas de Ataides <lucas.deataidesbarreto@windriver.com>
---
 cinder/templates/deployment-api.yaml | 1 +
 glance/templates/deployment-api.yaml | 2 ++
 heat/templates/deployment-api.yaml   | 1 +
 3 files changed, 4 insertions(+)

diff --git a/cinder/templates/deployment-api.yaml b/cinder/templates/deployment-api.yaml
index 3f4e941a..2f747db0 100644
--- a/cinder/templates/deployment-api.yaml
+++ b/cinder/templates/deployment-api.yaml
@@ -109,6 +109,7 @@ spec:
               scheme: {{ tuple "volume" "service" "api" . | include "helm-toolkit.endpoints.keystone_endpoint_scheme_lookup" | upper }}
               path: /
               port: {{ tuple "volume" "service" "api" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
+            initialDelaySeconds: 30
           livenessProbe:
             httpGet:
               scheme: {{ tuple "volume" "service" "api" . | include "helm-toolkit.endpoints.keystone_endpoint_scheme_lookup" | upper }}
diff --git a/glance/templates/deployment-api.yaml b/glance/templates/deployment-api.yaml
index 956639dc..a3405526 100644
--- a/glance/templates/deployment-api.yaml
+++ b/glance/templates/deployment-api.yaml
@@ -130,11 +130,13 @@ spec:
               scheme: HTTP
               path: /
               port: {{ tuple "image" "service" "api" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
+            initialDelaySeconds: 30
           livenessProbe:
             httpGet:
               scheme: HTTP
               path: /
               port: {{ tuple "image" "service" "api" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
+            initialDelaySeconds: 30
           volumeMounts:
             - name: pod-tmp
               mountPath: /tmp
diff --git a/heat/templates/deployment-api.yaml b/heat/templates/deployment-api.yaml
index 178bedda..8f621e0e 100644
--- a/heat/templates/deployment-api.yaml
+++ b/heat/templates/deployment-api.yaml
@@ -86,6 +86,7 @@ spec:
               scheme: {{ tuple "orchestration" "service" "api" . | include "helm-toolkit.endpoints.keystone_endpoint_scheme_lookup" | upper }}
               path: /
               port: {{ tuple "orchestration" "service" "api" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
+            initialDelaySeconds: 30
           livenessProbe:
             httpGet:
               scheme: {{ tuple "orchestration" "service" "api" . | include "helm-toolkit.endpoints.keystone_endpoint_scheme_lookup" | upper }}
-- 
2.25.1

